From 98f28e35ea383b47657f30ea342544cbff988adb Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 5 Apr 2024 09:47:20 +0200
Subject: [PATCH 581/841] refactor(stm32mp2): update reset_backup_domain
 function

Change the condition of the backup domain reset to check the
RCC_BDCR_RTCCKEN bit. When LSE CSS is enabled and a tamper event
occurs, the RTCSRC[1:0] bit-field is cleared, leading to a incorrect
cold boot detection.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 752fe0f0b36b47215a6f99ffb1371f5519b299d0 refactor(stm32mp2): update reset_backup_domain function]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I2fe7ab9f6d9fe0ba8078f782dd3251565e4bed47
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/371576
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/373931
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index 601fa1283..537e10daf 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -248,7 +248,7 @@ static void reset_backup_domain(void)
 #endif /* STM32MP21 */
 
 	/* Reset backup domain on cold boot cases */
-	if ((mmio_read_32(rcc_base + RCC_BDCR) & RCC_BDCR_RTCSRC_MASK) == 0U) {
+	if ((mmio_read_32(rcc_base + RCC_BDCR) & RCC_BDCR_RTCCKEN) == 0U) {
 		mmio_setbits_32(rcc_base + RCC_BDCR, RCC_BDCR_VSWRST);
 
 		while ((mmio_read_32(rcc_base + RCC_BDCR) & RCC_BDCR_VSWRST) == 0U) {
-- 
2.47.2

